pkg/scheduler/framework/plugins/nodeports/node_ports.go

```diff
@@ -76,6 +78,10 @@ func getContainerPorts(pods ...*v1.Pod) []*v1.ContainerPort {
 // PreFilter invoked at the prefilter extension point.
 func (pl *NodePorts) PreFilter(ctx context.Context, cycleState *framework.CycleState, pod *v1.Pod) (*framework.PreFilterResult, *framework.Status) {
 	s := getContainerPorts(pod)
```

Comment:
We can optimize further by making getContainerPorts only return ports with host port set, this check is currently done in the filter via https://github.com/kubernetes/kubernetes/blob/552f9b9dc6ee86574411b53759176b7544a2d314/pkg/scheduler/framework/types.go#L891